### DISCUSSION TITLE
Feature request: make DNSName overwriteable for certificate check

### DISCUSSION THREAD

**Discussion by km-manzm** [FreeOpcUa/opcua-asyncio#1485] [Ideas]
I am using the library in a dockerized Environment.
For the secure connection to OPC UA i use a certificate that is valid for the Host where the docker container is running.
However if i run my program in a docker container with this certificate, i get the following error message:

> certificate does not contain the hostname in DNSNames 79ec9126677a. Some applications will check this.
This happens because asyncua will check the hostname with the dnsname in the certificate in the check_certificate method of uacrypto.py. It will fail, because the docker container doesn't have the same dnsname as the host.

![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/60424464/a91801f0-0c65-4ba1-b530-c3cf5f172c3a)

My Suggestion would be that the hostname can be set manually (like it is already done with the application_uri)

**Comment by AndreasHeine** [comment]
generally speaking each OPC UA Client/Server has a ApplicationInstance-Certificate which is valid for only the instance! You need to set the hostname for the docker-container and set the proper hostname in the certificate! you need to look at it similar to virtual machines, so each container is a own machine and the "docker-host" is just the hardware which the software runs on.

**Reply by km-manzm** [reply]
Thanks for your response, that makes sense to me. However this is not possible in my setup because these settings are not in my hand.
Therefore it is fully understandable if my suggestions won't be implemented due to the security issues.
